docs/DeveloperGuide.adoc

```diff
@@ -74,42 +70,191 @@ The sections below give more details of each component.
 [[Design-Ui]]
 === UI component
 
-.Structure of the UI Component
+.Main structure of the UI Component
```

Comment:
The UI architecture diagram looks well drawn and clear!
- Uses the correct UML notation.
- Not cluttered with too much unnecessary details
- Each diagram is relatively simple and easy to understand